### PR TITLE
CES-11261 updated assign_role ...

### PR DESCRIPTION
**Pull request by gaelrehault**
- to handle missing attribute prior to raid/bios config ( causing errors during cleaning )
- removed unused code.

### CODE REVIEW

**Comment by heydog15** [comment]
This code has been deployed and tested successfully on a few racks.

File: src/pilot/assign_role.py [hunk 1]
```diff
@@ -778,32 +778,6 @@ def configure_raid(ironic_client, node_uuid, super_role, os_volume_size_gb,
 
     # Work around the bugs in the ironic DRAC driver's RAID clean steps.
```
Comment (rpioso) [inline src/pilot/assign_role.py:780]: Let us delete these two (2) lines, too.

File: src/pilot/assign_role.py [hunk 2]
```diff
@@ -817,35 +791,6 @@ def configure_raid(ironic_client, node_uuid, super_role, os_volume_size_gb,
...
-        node_uuid, ironic_client, drac_client, 'RAID',
-        controllers_to_physical_disk_ids)
-
-    if succeeded:
-        LOG.info("Completed converting physical disks configured to back RAID "
-                 "logical disks to RAID mode")
-    else:
-        LOG.critical("Attempt to convert physical disks configured to back "
-                     "RAID logical disks to RAID mode failed")
-        return False
```
Comment (rpioso) [inline src/pilot/assign_role.py:794]: Please delete this blank line, too. My Python-aware Eclipse PyDev editor complains loudly about extra lines.

**Review by rpioso — COMMENTED** [review]
I have one (1) major comment and a couple of minor ones.

Both @cdearborn and I believe the code in the `assign_role()` function which ensures `boot_mode:uefi` is in the `capabilities` property on the ironic bare metal node object should be unconditionally executed. Presently, it is conditioned on the presence of `capabilities` in `node.properties`. Any reason why that is?

Our gut is that it happens to work, because by the time it runs, the node has already been in-band introspected and has `capabilities`. Fortunate :-)

The block of code I am referring to is in `assign_role.py`, lines 849-855. For your convenience, I have copied it below.

    if 'capabilities' in node.properties:
        value = "{},{},boot_mode:uefi,boot_option:local".format(role, node.properties['capabilities'])
        LOG.info(str(node.properties))
        LOG.info(str(value))
    else:
        value = "{},boot_option:local".format(role)
        LOG.info("..!")

If necessary, this code could be merged as-is and a follow-up PR submitted to address my comments. I would prefer they be addressed now. Your call, though.

**Comment by gaelrehault** [comment]
Addressed the minor comments. 

In regard to "_Presently, it is conditioned on the presence of capabilities in node.properties. Any reason why that is?_", I can only point you to the following PR, reviewed and merge by yourself nearly 2 years ago , that introduced the condition in the first place : 
https://github.com/dsp-jetpack/JetPack/pull/183 

If you feel this should now be questioned & removed, i m happy to take it out, but this isn't something that was recently introduced as you can see from the above; 
Given the time constraint we re under, I would rather address it at a later stage, but leaving that decision up to @heydog15 .

**Comment by gaelrehault** [comment]
Richard, addressed the minor issues. 
In relation to the capabilities, would you mind clarifying a little more ? what i m understanding you re saying, is that capabilities will only be available if introspection ran in-band, but not out of band ? If that s the case we should add the property in both cases ?

**Comment by heydog15** [comment]
We need both in-band and out of band either way.  Until we can do out of band NFV stuff, we need both.

**Comment by rpioso** [comment]
> Richard, addressed the minor issues.
> In relation to the capabilities, would you mind clarifying a little more ? what i m understanding you re saying, is that capabilities will only be available if introspection ran in-band, but not out of band ? If that s the case we should add the property in both cases ?

Let me try again. The `if` statement is not needed. Just do it! (tm) without the `if` statement and `else` clause.

    value = "{},{},boot_mode:uefi,boot_option:local".format(role, node.properties['capabilities'])
    LOG.info(str(node.properties))
    LOG.info(str(value))

Further clarifying, `capabilities` is available if in-band introspection (IB) **or** out-of-band (OOB) inspection is performed before `assign_role.py` is executed, which I believe it is. Agree? If so, that is why the code has been working as-is. If the order were reversed, it would not work. So, the current code is not robust, confusing, and difficult to maintain.

I understand IB is presently used by JetStrem, because of the need for NUMA topology information, which OOB does not presently offer. Should that information be added to OOB and JetStream changed to use OOB, my suggestion would continue to work.

**Comment by gaelrehault** [comment]
Removed condition , and added missing commit required

File: src/pilot/assign_role.py [hunk 3]
```diff
@@ -901,26 +846,16 @@ def assign_role(ip_mac_service_tag, node_uuid, role_index, os_volume_size_gb,
...
-    if 'capabilities' in node.properties:
-        value = "{},{},boot_mode:uefi,boot_option:local".format(role, node.properties['capabilities'])
-        LOG.info(str(node.properties))
-        LOG.info(str(value))
-    else:
-        value = "{},boot_option:local".format(role)
-        LOG.info("..!")
+    value = "{},{},boot_mode:uefi,boot_option:local".format(role, node.properties['capabilities'])
+    LOG.info(str(node.properties))
+    LOG.info(str(value))
```
Comment (rpioso) [inline src/pilot/assign_role.py:851]: As a follow-up to this pull request, let's make this work when out-of-band (OOB) inspection or in-band (IB) introspection has not been performed before `assign_role.py` is run. That workflow is used by the Ironic Team when integration testing their changes. In that case, the `capabilities` property may not be present and line 849 above would cause an exception to be raised.

The new implementation could be something like the following:

    if 'capabilities' in node.properties:
        value = "{},{},boot_mode:uefi,boot_option:local".format(role, node.properties['capabilities'])
    else:
         value = "{},boot_mode:uefi,boot_option:local".format(role)

    LOG.info(str(node.properties))
    LOG.info(str(value))

**Review by rpioso — COMMENTED** [review]
LGTM! Thank you, @gaelrehault and @himanshusingh2!

I suggest we create a follow-up pull request to address a shortcoming which the Ironic Team would likely encounter when using `assign_role.py` as an integration test tool. It does not affect JetStream nor automated deployment.
